app/models/doi.rb

```diff
@@ -1765,6 +1765,36 @@ def self.transfer(options={})
     response.results.total
   end
 
+  def self.loop_through_dois(options)
+    size = (options[:size] || 1000).to_i
+    cursor = [options[:from_id], options[:until_id]]
+    filter = options[:filter] || {}  
+    label = options[:label] || "" 
+    job_name = options[:job_name] || "" 
+    query = options[:query] || nil
+
+    query_hsh = {page: { size: 1, cursor: [] } }.merge(filter)
+
+    response = Doi.query(query, query_hsh)
+    Rails.logger.warn "#{label} #{response.results.total} Dois with #{label}."
+
+    # walk through results using cursor
+    if response.results.total.positive?
+      while response.results.results.length.positive?
+        response = Doi.query(query, query_hsh.merge!(page: { size: size, cursor: cursor }))
+        break unless response.results.results.length.positive?
+
+        Rails.logger.warn "#{label} #{response.results.results.length}  Dois starting with _id #{response.results.to_a.first[:_id]}."
+        cursor = response.results.to_a.last[:sort]
+        Rails.logger.warn "#{label} Cursor: #{cursor} "
```

Comment:
Please limit `logger.warn` messages to a minimum, as they are sent to DataDog. Maybe `logger.info` for situations where we need logging.